database.go

```diff
@@ -160,6 +163,11 @@ func (db *BadgerDB) Close() error {
 	return nil
 }
 
+// Subscribe to watch for changes for the given prefixes
+func (db *BadgerDB) Subscribe(ctx context.Context, cb func(kv *KVList) error, prefixes []byte) error {
+	return db.db.Subscribe(ctx, cb, prefixes)
```

Comment:
maybe you could update this function to return a channel, and define the callback inside this function, and when the callback is called then put the key-values into the channel? you would need to define some non-badger key-value type. this would avoid the need to pass in the callback

I noticed that badger.Subscribe blocks until the context is done: https://pkg.go.dev/github.com/dgraph-io/badger/v2#DB.Subscribe  so maybe you want to start `db.Subscribe` in a goroutine?